pkg/selectors/selectors.go

```diff
@@ -5,6 +5,14 @@ package selectors
 
 import (
 	"encoding/binary"
+	"sync"
+)
+
+// as we use a single names_map for all kprobes, so we have to use
+// a global variable to assign values to binary names
+var (
+	binMu  sync.Mutex
```

Comment:
Usually a mutex should be part of the context it locks (struct fields...), so we know the locking

Reply:
ah it is locking binIdx below hmm ok

Reply:
Yes, this is for a global variable.

Reply:
ok , see my other comments about using the same lock in an other context